tiledb/api/src/non_empty_domain.rs

```diff
@@ -0,0 +1,111 @@
+use std::fmt::{Debug, Formatter, Result as FmtResult};
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+use crate::datatype::Datatype;
+
+#[derive(Serialize, Deserialize)]
```

Comment:
Probably add `Clone` to the derive lists, can't hurt to have and might be useful to somebody

Reply:
Done.